2020/day-16/arora-aditya/day16.py

```diff
@@ -0,0 +1,128 @@
+from typing import List, Dict, Set
+from collections import *
+from functools import lru_cache
+from math import log10
+from helper.submit.submit import *
+from utils import *
+from pprint import pprint as pp
+
+
+DAY = 16
+setup(DAY)
+lines = read_file(DAY)
+
+
+# Part 1
+##################################################
+def get_invalids(di, nums):
+    invalids = []
+    for num in nums:
+        flag = False
+        for x in di.values():
+            if num in x:
+                flag = True
+                break
+        if not flag:
+            invalids.append(num)
+    return invalids
+
+def part1(lines: List[str]) -> int:
+    i = 0
+    line = lines[i]
+    di = defaultdict(set)
+    while len(line) != 0:
```

Comment:
Ya, I hardcoded my parsing. Def paid off for p1. Not great for testing the example though.